### PR TITLE
feature: allowed IPv6 address with square brackets in parse_ipv6 method

### PR DESCRIPTION
**Pull request by spacewander**

### CODE REVIEW

File: resty/ipmatcher.lua [hunk 1]
```diff
@@ -55,6 +56,11 @@ local function parse_ipv6(ip)
         return false
     end
 
+    if str_byte(ip, 1, 1) == str_byte('[') then
```
Comment (membphis) [inline resty/ipmatcher.lua]: `[::1x` this a invalid IPv6 address.

**Comment by spacewander** [comment]
@membphis 
Updated.

**Comment by membphis** [comment]
@spacewander Do you think I need to release a new version now? Then we can use the new version at APISIX.

**Comment by spacewander** [comment]
A new version is need to make APISIX work with IPv6 correctly.

**Comment by membphis** [comment]
Already done, the new version is  `0.3` .

https://luarocks.org/modules/membphis/lua-resty-ipmatcher

**Pull request merged**